src/Core/Domain/Address/Command/AbstractEditAddressCommand.php

```diff
@@ -362,7 +362,7 @@ public function getHomePhone(): ?string
     }
 
     /**
-     * @param string|null $homePhone
+     * @param string $homePhone
```

Comment:
Same here. We made these fields nullable to allow people to perform partial updates. Nullable fields are valid for Edit Commands (but NOT for Create Commands)

Reply:
@matks Sorry ! As `declare(strict_types=1);` is defined, a nullable value in parameter will trigger a fatal error :

![image](https://user-images.githubusercontent.com/1533248/89200943-0ea76300-d5b1-11ea-9409-c7d984ad905c.png)

Reply:
@Progi1984 We dont call the setter so the value stays null because it was initialized `null` when we call `$command = new AddressCommand()`. So there is no fatal error 😉

Reply:
So the setter must be `string`. The getter and the property must be `string|null`. Are we agree ?

Reply:
I'm not sure 🤔  .
Imagine the following scenario:
- merchant inputs a `homePhone` data
- so form is populated with this data, and EditAddressCommand is populated with this data
- however merchant has installed a module that handles phone numbers in a specific ERP using an API !
- so the module wants to empty the Core field and put the data in its own database using an API call

If you put `string` mandatory in the setter, we cannot empty a field after it has been set 🤔  consequently limiting the ability of users to manipulate it.

A workaround is to create a ModuleExtendedEditAddressCommand to be used 😄  but it's a little bit more work

Reply:
But if we change to null, we need to change the signature. It become a BC. So it must be done in another PR because it's not the scope of the PR.

Reply:
Do you mean **this** is a BC ?
```
    public function setHomePhone(string $homePhone): self
```
⬇️ 
```
    public function setHomePhone(string $homePhone = null): self
```

Since that change is backward-compatible with previous version it's not a BC break IMO 🤔  . BC Break stands for "Backward Compatibility Change". Signature changes are allowed provided they are backward compatible. That is how it sounds to me.

I however agree this PR is getting out-of-scope 👍  so I suggest to make phpstan ignore this AbstractEditAddressCommand file for now, and then in another PR, we can sort it out by finding the best solution 🎖️

Reply:
@matks changing the signature is a BC. Since PHP can be used as strict mode, and be able to catch `Error`, you create a new behavior here.

Reply:
I think we should change the signature when the need will be there.

Reply:
I think this must be decided by @eternoendless . Perosnally, I am 50% against ; 50% for.

I'm 50% for because it's clearer 😄 
I'm 50% against because of the "X wants to empty the field" usecase" described here: https://github.com/PrestaShop/PrestaShop/pull/20118#discussion_r464846373